lib/uwb/include/uwb/UwbSessionEventCallbacks.hxx

```diff
@@ -41,67 +41,51 @@ enum class UwbSessionEndReason {
 /**
  * @brief Interface for receiving events from a UwbSession. This is the primary
  * method to receive information from near peers.
- * 
- * A pointer to the UwbSession for which the callbacks are bound is provided in
- * each callback. This enables using the same callback object instance with
- * multiple sessions where the pointer value can be used to lookup the
- * associated session for the event. This pointer must not be cached nor used
- * beyond execution of the callback.
+ *
  */
 struct UwbSessionEventCallbacks
```

Comment:
Omitting the `UwbSession` pointer sacrifices the caller's ability to specify a single set of callbacks for all instances they wish to register since the session pointer disambiguates which session the callback is for. We should restore this argument in all the callback functions here.

This goes back to the design challenge you faced earlier, in that the connector has no knowledge of a `UwbSession` (which is good) so it can't easily determine the `UwbSession*` pointer to pass. The dilemma of ownership of lambdas is yet another design challenge and complication. Together, I think these issues reveal that the callbacks invoked by the `UwbDeviceConnector` should be a completely separate entity from those invoked by `UwbSession` and `UwbDevice` since they're providing different functionality. The connector deals in session ids, the `UwbDevice` and `UwbSession` deal in `UwbSession` object instances.

I think the way you've defined the callbacks now (with lambdas) are perfect for `UwbDeviceConnector` because `UwbDeviceConnector` is always meant to be wholly owned by something, and the way they were defined previously were suitable for `UwbDevice` and `UwbSession`. So, I think the best solution is to restore the old callbacks, and take the new ones you've defined and make them specific to `UwbDeviceConnector`.

Reply:
I reverted the UwbDeviceEventCallbacks and UwbSessionEventCallbacks to how they were before and added the UwbRegisteredCallbacks which will what the UwbDeviceConnector use to execute those callbacks.